src/Symfony/Component/Messenger/Bridge/MongoDB/Transport/TransportFactory.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\MongoDB\Transport;
+
+use MongoDB\Database;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Messenger\Bridge\MongoDB\Extension\DocumentEnhancer;
+use Symfony\Component\Messenger\Exception\InvalidArgumentException;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+use Symfony\Component\Messenger\Transport\TransportFactoryInterface;
+
+/**
+ * @template-implements TransportFactoryInterface<MongoDbTransport>
+ */
+final class TransportFactory implements TransportFactoryInterface
+{
+    private const array DEFAULT_OPTIONS = [
+        self::COLLECTION_NAME => 'messenger_messages',
+        self::QUEUE_NAME => 'default',
+        self::REDELIVER_TIMEOUT => 3_600,
+        self::DOCUMENT_ENHANCERS => [],
+    ];
+
+    public const string CONNECTION_NAME = 'connection_name';
+
+    public const string COLLECTION_NAME = 'collection_name';
+
+    public const string QUEUE_NAME = 'queue_name';
+
+    public const string REDELIVER_TIMEOUT = 'redeliver_timeout';
+
+    public const string DOCUMENT_ENHANCERS = 'document_enhancers';
+
+    public function __construct(private readonly ContainerInterface $container)
+    {
+    }
+
+    /**
+     * @param array<string, mixed> $options
+     */
+    public function supports(string $dsn, array $options): bool
+    {
+        return str_starts_with($dsn, 'mongodb://');
```

Comment:
`mongodb+srv` must be also supported. It's a protocol that read the DNS SRV entries to get replicaset members.

```suggestion
        return str_starts_with($dsn, 'mongodb://') || str_starts_with($dsn, 'mongodb+srv://');
```